binary_transparency/firmware/integration/ft_test.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+set -e
+. $(go list -f '{{.Dir}}' github.com/google/trillian)/integration/functions.sh
+INTEGRATION_DIR="$( cd "$( dirname "$0" )" && pwd )"
+. "${INTEGRATION_DIR}"/ft_functions.sh
+
+ft_prep_test
+
+function cleanup {
+banner "Cleaning up"
+ft_stop_test
+TO_KILL=()
+}
+
+trap cleanup EXIT
+
+DEVICE_STATE=$(mktemp -d /tmp/dummy-XXXXX)
+UPDATE_FILE=$(mktemp /tmp/update-XXXXX.json)
+
+
+# Cleanup for the Trillian components
+TO_DELETE="${TO_DELETE} ${ETCD_DB_DIR}"
+TO_KILL+=(${LOG_SIGNER_PIDS[@]})
+TO_KILL+=(${RPC_SERVER_PIDS[@]})
+TO_KILL+=(${ETCD_PID})
+
+# Cleanup for the personality
+TO_DELETE="${TO_DELETE} ${FT_CAS_DB} ${DEVICE_STATE} ${UPDATE_FILE} ${FT_MONITOR_LOG}"
+TO_KILL+=(${FT_SERVER_PID})
+TO_KILL+=(${FT_MONITOR_PID})
+
+echo "Running test(s)"
+pushd "${INTEGRATION_DIR}"
+
+banner "Logging initial firmware"
+go run ../cmd/publisher/ \
```

Comment:
It might be worth adding the timestamp flag. I think that would be sufficient to at least make the tests deterministic in terms of generated data.

Reply:
done